src/praisonai-agents/tests/unit/tools/test_non_idempotent_tool_not_retried.py

```
@@ -0,0 +1,72 @@
+        raise RuntimeError(message)
+
+    if idempotent is not None:
+        charge_card.idempotent = idempotent
+
+    cfg = ToolConfig(retry_policy=RetryPolicy(max_attempts=max_attempts,
+                                              initial_delay_ms=0))
+    agent = Agent(instructions="x", tools=[charge_card], tool_config=cfg)
+    try:
+        agent.execute_tool("charge_card", {"amount": "100"})
+    except Exception:
+        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)